pkg/config/config.go

```diff
@@ -441,7 +453,7 @@ func stringReverse(s string) string {
 	return string(chars)
 }
 
-func (c *Config) validateDomainNames() error {
+func (c *BaseConfig) ValidateDomainNames() error {
```

Comment:
Since this file is quite big already - 
How about moving BaseConfig to another file?

Reply:
@itaigilo - IMO the file is not big enough to justify splitting (644 lines long)